src/components/profile/HeatmapCalendar.tsx

```diff
@@ -0,0 +1,184 @@
+import styled from '@emotion/styled';
+import { useEffect, useRef } from 'react';
+import CalendarHeatmap from 'react-calendar-heatmap';
+import { Tooltip } from 'react-tooltip';
+import 'react-tooltip/dist/react-tooltip.css';
+import { getLastYearDate } from 'utils';
+
+// TODO: Mock data 제거
+const data = [
+  {
+    date: '2022-04-25',
+    count: 1,
+  },
+  {
+    date: '2023-01-01',
+    count: 1,
+  },
+  {
+    date: '2023-01-11',
+    count: 2,
+  },
+  {
+    date: '2023-02-01',
+    count: 3,
+  },
+  {
+    date: '2023-04-24',
+    count: 3,
+  },
+  {
+    date: '2024-03-03',
+    count: 6,
+  },
+];
+
+// TODO: constants/common으로 이동
+const WEEKDAY = ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'];
+
+interface Value {
+  date: string;
+  count: number;
+}
+
+export const HeatmapCalendar = () => {
+  const today = new Date();
+
+  const boxRef = useRef<HTMLDivElement | null>(null);
+
+  const getClassForValue = (value: Value) => {
+    if (value === null) return 'color-step-0';
+
+    const { count } = value;
+
+    if (count > 0) return 'color-step-1';
+    if (count > 2) return 'color-step-2';
+    if (count > 4) return 'color-step-3';
+
+    return 'color-step-0';
+  };
+
+  // TODO: Tooltip 필요한지 확인
+  const getTooltipDataAttrs = (value: Value) => {
+    const content =
+      value.date === null ? today : `${value.date} has count: ${value.count}`;
+
+    return {
+      'data-tooltip-id': 'my-tooltip',
+      'data-tooltip-content': content,
+    };
+  };
+
+  const handleClick = (value: Value) => {
+    if (value === null) return;
+
+    // TODO: 클릭 시 해당 날짜 활동 내역 보여주기
+    console.log(`Clicked on value with count: ${value.count}`);
+  };
+
+  useEffect(() => {
+    if (boxRef?.current !== null) {
+      boxRef.current.scrollTo({ left: 1520 });
```

Comment:
히트맵이 1년의 데이터를 나타내면서 스크롤이 생깁니다.
따라서, 최신 데이터를 먼저 보여주기 위해 scrollTo를 이용했습니다.
left 값은 히트맵의 너비값입니다.

Reply:
p3;

코멘트 달아주신 의미대로 변수를 만들어서 사용하면 좋을 것 같습니다~
아래는 예시를 위해 써봤는데 변수명은 바꾸셔도 무방합니다!

```suggestion
const HEATMAP_BOX_WIDTH = 1520
if (boxRef?.current !== null) {
      boxRef.current.scrollTo({ left: HEATMAP_BOX_WIDTH });
}
```

Reply:
p3;

코멘트 달아주신 의미대로 변수를 만들어서 사용하면 좋을 것 같습니다~
아래는 예시를 위해 써봤는데 변수명은 바꾸셔도 무방합니다!

```suggestion
const HEATMAP_BOX_WIDTH = 1520
if (boxRef?.current !== null) {
      boxRef.current.scrollTo({ left: HEATMAP_BOX_WIDTH });
}
```

Reply:
@sohyeonAn 
오! 너무 좋아요! 감사합니다!